File: llm_generated_code/ChatGPT/Python/Level1/Q36.py
def min_operations_to_equal_elements(n, a):
    freq = {}
    for num in a:
        freq[num] = freq.get(num, 0) + 1
    max_freq = max(freq.values())
    if max_freq == n:
        return 0
    operations = 0
    current_copies = max_freq
    while current_copies < n:
        operations += 1
        to_add = min(current_copies, n - current_copies)
        current_copies += to_add
        operations += to_add
    return operations

t = int(input())
results = []
for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    results.append(min_operations_to_equal_elements(n, a))

for result in results:
    print(result)

File: llm_generated_code/ChatGPT/CPP/Level1/Q36.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>

using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;
        
        vector<int> a(n);
        unordered_map<int, int> freq;
        
        for (int i = 0; i < n; i++) {
            cin >> a[i];
            freq[a[i]]++;
        }
        
        int maxFreq = 0;
        for (auto &entry : freq) {
            maxFreq = max(maxFreq, entry.second);
        }
        
        int operations = 0, remaining = n - maxFreq;
        while (remaining > 0) {
            operations++;          // Clone operation
            int increase = min(maxFreq, remaining);
            operations += increase; // Swap operations
            maxFreq += increase;
            remaining -= increase;
        }
        
        cout << operations << endl;
    }

    return 0;
}
